### PR TITLE
Respond with information on VM revert

### PR DESCRIPTION
**Pull request by gtrepta**
fixes: https://github.com/runtimeverification/firefly/issues/322

### CODE REVIEW

File: web3.md [hunk 1]
```diff
@@ -1182,6 +1199,11 @@ Transaction Receipts
          <k> #halt ~> #catchHaltTx ACCT => #mkCodeDeposit ACCT ... </k>
       requires ACCT =/=K .Account
 
+    syntax KItem ::= "#setPC" Int
+ // -----------------------------
+    rule <k> #setPC PCOUNT => . ... </k>
+         <pc> _ => PCOUNT </pc>
+
```
Comment (ehildenb) [inline web3.md]: It feels hacky to set the program counter to get the returned opcode position. Are there any other ways we can do this? We could, for example, change `EVMC_REVERT` to store the program counter that the revert happens at (similarly for `EVMC_INVALID`).

Comment (gtrepta) [reply web3.md]: This is similar to the use of `#refund` to preserve the gas cell, so it seemed ok to me. I don't see the `EVMC_INVALID` example you're talking about, can you point it to me?

File: web3.md [hunk 2]
```diff
@@ -285,11 +286,13 @@ WEB3 JSON RPC
          <batch> [ _ ] </batch>
 
     syntax KItem ::= #rpcResponseSuccess       ( JSON                )
+                   | #rpcResponseRevert        ( JSON , JSON         )
```
Comment (ehildenb) [inline web3.md]: Since this doesn't do anything specific to `Revert`, let's rename it to `#rpcResponseSuccessWithError`.

File: web3.md [hunk 3]
```diff
@@ -284,12 +285,14 @@ WEB3 JSON RPC
...
-    syntax KItem ::= #rpcResponseSuccess       ( JSON                )
-                   | #rpcResponseError         ( Int , String        )
-                   | #rpcResponseError         ( Int , String , JSON )
-                   | #rpcResponseUnimplemented ( String              )
+    syntax KItem ::= #rpcResponseSuccess          ( JSON                )
+                   | #rpcResponseSuccessException ( JSON , JSON         )
+                   | #rpcResponseError            ( Int , String        )
+                   | #rpcResponseError            ( Int , String , JSON )
+                   | #rpcResponseUnimplemented    ( String              )
  // ------------------------------------------------------------------
```
Comment (ehildenb) [inline web3.md]:
```suggestion
 // ---------------------------------------------------------------------
```

File: web3.md [hunk 4]
```diff
@@ -284,12 +285,14 @@ WEB3 JSON RPC
...
+                   | #rpcResponseSuccessException ( JSON , JSON         )
+                   | #rpcResponseError            ( Int , String        )
+                   | #rpcResponseError            ( Int , String , JSON )
+                   | #rpcResponseUnimplemented    ( String              )
+ // ---------------------------------------------------------------------
     rule <k> #rpcResponseSuccess(J)             => #sendResponse( "result" : J )                                                ... </k> requires isProperJson(J)
+    rule <k> #rpcResponseSuccessException(RES, ERR)       => #sendResponse( ( "result" : RES, "error": ERR ) )                            ... </k> requires isProperJson(RES) andBool isProperJson(ERR)
     rule <k> #rpcResponseError(CODE, MSG)       => #sendResponse( "error" : { "code": CODE , "message": MSG } )                 ... </k>
     rule <k> #rpcResponseError(CODE, MSG, DATA) => #sendResponse( "error" : { "code": CODE , "message": MSG , "data" : DATA } ) ... </k> requires isProperJson(DATA)
     rule <k> #rpcResponseUnimplemented(RPCCALL) => #sendResponse( "unimplemented" : RPCCALL )                                   ... </k>
```
Comment (ehildenb) [inline web3.md]: Please fix the formatting here.

**Pull request merged**